spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2864Test.java

```diff
@@ -0,0 +1,37 @@
+package edu.umd.cs.findbugs.detect;
+
+import static edu.umd.cs.findbugs.test.CountMatcher.containsExactly;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+import edu.umd.cs.findbugs.AbstractIntegrationTest;
+import edu.umd.cs.findbugs.BugCollection;
+import edu.umd.cs.findbugs.test.SpotBugsRunner;
+import edu.umd.cs.findbugs.test.SpotBugsExtension;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcher;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.nio.file.Paths;
+
+/**
+ * @author gtoison
+ *
+ */
+@ExtendWith(SpotBugsExtension.class)
+class Issue2864Test extends AbstractIntegrationTest {
+
+    @Test
+    void testIssue(SpotBugsRunner spotbugs) {
+        BugCollection bugCollection = spotbugs.performAnalysis(
+                Paths.get("../spotbugsTestCases/src/classSamples/classEnhanceByHibernate/HibernateEnhancedEntity.class"));
```

Comment:
When running this test, several classes are missing:
```
The following classes needed for analysis were missing:
  org.hibernate.engine.spi.ManagedEntity
  org.hibernate.engine.spi.ExtendedSelfDirtinessTracker
  org.hibernate.bytecode.enhance.internal.tracker.SimpleFieldTracker
  org.hibernate.bytecode.enhance.internal.tracker.SimpleCollectionTracker
  org.hibernate.engine.spi.EntityEntry
  org.hibernate.bytecode.enhance.internal.tracker.DirtyTracker
  org.hibernate.bytecode.enhance.spi.interceptor.LazyAttributeLoadingInterceptor
  org.hibernate.bytecode.enhance.spi.CollectionTracker
  org.hibernate.collection.spi.PersistentCollection
```

I'm not sure, how much these classes are really needed, and what exactly causes when there are missing classes. The PR solves the referenced issue, so maybe it's not a big deal in this case.

Reply:
The classes references are from the bytecode added by Hibernate's enhancer.
I looked into adding Hibernate as a dependency and also adding the enhancer plugin to the build but that seems to be overkill, so I just compiled a class and copied it into the samples folder.